toys/rocks-candy-shop/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+use scrypto::prelude::*;
+
+/* This a non-trivial example that shows off a blueprint that operates with
+    with an adjustable number of vaults. My main discover was that you can
+    return Vec<Bucket> and the resim methd call can handle it and update
+    the account correctly.
+
+   If someone wants to do so there are many possible extensions:
+        - add purchase method(s) via XRD
+            - have the purchase method(s) also return change
+            - add badges to control who can get the collected XRD
+        - allow for different prices for each type of candy
+        - allow candy restocking by a badged account (hint: mutable supply)
+        - tag the candy_vaults with the rri instead of the token symbol
+        - many more...
+*/
+
+blueprint! {
+    struct CandyShop {
+        // The different kinds of candies are kept here with each in a tuple
+        // with a unique tag string that doubles as the candies' token symbol.
+        candy_vaults: Vec<(String,Vault)>
```

Comment:
I would consider using a `HashMap` or `LazyMap` to make vault retrieval easier.

Reply:
I have a separate example called Vending Machine that uses a HashMap of Vaults and so I think that I will leave this one as is just to show a different way to handle a requirement for dynamic Vaults. I will look into Option<Bucket> and fit that into an example somewhere as well. The empty_bucket approach shown is going to be a common idiom I would think simply because of it's simplicity.